argocd/utils.go

```diff
@@ -63,6 +63,16 @@ func expandStringList(l []interface{}) (result []string) {
 	return
 }
 
+func sliceOfString(slice []interface{}) []string {
+	result := make([]string, len(slice))
+
+	for i, s := range slice {
+		result[i] = s.(string)
+	}
+
+	return result
+}
+
```

Comment:
Panics may happen, better be safe than sorry!

```suggestion
func sliceOfString(slice []interface{}) ([]string, error) {
	result := make([]string, len(slice))

	for i, s := range slice {
		switch s.(type) {
	        case string:
			result[i]= s.(string)
	        default:
	        	return nil, fmt.Errorf("element %v was not of type string: %T", s, s)
	        }
	}

	return result, nil
}

```